pro_wes/ga4gh/wes/workflow_runs.py

```diff
@@ -136,25 +145,34 @@ def run_workflow(
             if response.status_code == 403:
                 raise Forbidden
             raise InternalServerError
-        document_stored: DbDocument = db_connector.upsert_fields_in_root_object(
+        updated_document_stored: DbDocument = db_connector.upsert_fields_in_root_object(
             root="wes_endpoint",
             run_id=response.run_id,
         )
 
+        if updated_document_stored.wes_endpoint is None:
+            raise ValueError("No WES endpoint available.")
+
         # track workflow progress in background
         task__track_run_progress.apply_async(
             None,
             {
                 "jwt": kwargs.get("jwt", None),
-                "remote_host": document_stored.wes_endpoint.host,
-                "remote_base_path": document_stored.wes_endpoint.base_path,
-                "remote_run_id": document_stored.wes_endpoint.run_id,
+                "remote_host": updated_document_stored.wes_endpoint.host,
+                "remote_base_path": updated_document_stored.wes_endpoint.base_path,
+                "remote_run_id": updated_document_stored.wes_endpoint.run_id,
             },
-            task_id=document_stored.task_id,
+            task_id=updated_document_stored.task_id,
             soft_time_limit=controller_config.timeout_job,
         )
 
-        return {"run_id": document_stored.run_log.run_id}
+        if updated_document_stored.wes_endpoint is None:
+            raise ValueError("No WES endpoint available.")
+
+        if updated_document_stored.wes_endpoint.run_id is None:
+            raise ValueError("WES endpoint does not have run_id.")
```

Comment:
I'm not too happy with all of these checks (here and above) but cannot think of a better solution wihtout diving into the code. Perhaps defining some validator method somewhere to check the content of the `.wes_endpoint` attribute? Then we'd just need to call that validator (which would either raise an error if any of the values were `None` or otherwise return nothing).

The question is whether mypy will scan deep enough to be satisfied with such an approach. If not, what we can always do if _we_ know for sure that, e.g., a given value can't really be `None` (because we have just checked it and mypy failed to process that), is to just add lines like `assert variable is not None`. It's a bit ugly, but it will satisfy mypy and has the advantage that we don't need to write unit tests to check a conditional branch for something that really can't/shouldn't happen.